Framework/Crystal/inc/MantidCrystal/FindSXPeaks.h

```diff
@@ -39,12 +37,29 @@ struct DLLExport SXPeak {
       throw std::invalid_argument(
           "SXPeak: Cannot have zero sized spectral list");
     }
-    if (Ltot < 0) {
+    if (!spectrumInfo.hasUniqueDetector(_wsIndex)) {
+      throw std::invalid_argument("SXPeak: Spectrum at ws index " +
+                                  std::to_string(wsIndex) +
+                                  " has more than a single detector");
```

Comment:
The test you added forbids detector groups (spectra with more than 1 detector), so unless that is the behavior of the original code you cannot add this test. Maybe a check for `spectrumInfo.hasDetectors(i)` would make sense.